### ISSUE TITLE
@codeCoverageIgnore not applied

### ISSUE THREAD

**Issue by lferro9000** [sebastianbergmann/php-code-coverage#426]
On the following code (trimmed for brevity):

    class datasource_children extends datasource_sql
    {
        // @codeCoverageIgnore
        protected function SQLUpdateBuild($table, $into, array &$data, $whereSql, $joinSQL = null) {}
    }

With the associated datasource_sql:

    abstract class datasource_sql extends datasource_dbe {
        // @codeCoverageIgnore
        abstract protected function SQLUpdateBuild($table, $into, array &$data, $whereSql, $joinSQL = null); // returns single SQL Statement
    }

Note: datasource_dbe has no relations with SQLUpdateBuild method.

When typing:

    $> phpunit --no-coverage

The tests run fine without any issue.

When typing:

    $> phpunit

The tests run fine, but code coverage crashes with the following error:

    Generating code coverage report in Clover XML format ...PHP Fatal error:  Declaration of datasource_children::SQLUpdateBuild() must be compatible with datasource_sql::SQLUpdateBuild($table, $into, array &$data, $whereSql, $joinSQL = NULL) in /Users/Shared/Jenkins/Home/workspace/dbe/dbEngine/test/Generator/SqlIncTestSqlInc.php on line 4
    PHP Stack trace:
    PHP   1. {main}() /usr/local/Cellar/phpunit/5.2.9/libexec/phpunit-5.2.9.phar:0
    PHP   2. PHPUnit_TextUI_Command::main() /usr/local/Cellar/phpunit/5.2.9/libexec/phpunit-5.2.9.phar:520
    PHP   3. PHPUnit_TextUI_Command->run() phar:///usr/local/Cellar/phpunit/5.2.9/libexec/phpunit-5.2.9.phar/phpunit/TextUI/Command.php:109
    PHP   4. PHPUnit_TextUI_TestRunner->doRun() phar:///usr/local/Cellar/phpunit/5.2.9/libexec/phpunit-5.2.9.phar/phpunit/TextUI/Command.php:158
    PHP   5. PHP_CodeCoverage_Report_Clover->process() phar:///usr/local/Cellar/phpunit/5.2.9/libexec/phpunit-5.2.9.phar/phpunit/TextUI/TestRunner.php:449
    PHP   6. PHP_CodeCoverage->getReport() phar:///usr/local/Cellar/phpunit/5.2.9/libexec/phpunit-5.2.9.phar/php-code-coverage/CodeCoverage/Report/Clover.php:43
    PHP   7. PHP_CodeCoverage_Report_Factory->create() phar:///usr/local/Cellar/phpunit/5.2.9/libexec/phpunit-5.2.9.phar/php-code-coverage/CodeCoverage.php:139
    PHP   8. PHP_CodeCoverage->getData() phar:///usr/local/Cellar/phpunit/5.2.9/libexec/phpunit-5.2.9.phar/php-code-coverage/CodeCoverage/Report/Factory.php:24
    PHP   9. PHP_CodeCoverage->addUncoveredFilesFromWhitelist() phar:///usr/local/Cellar/phpunit/5.2.9/libexec/phpunit-5.2.9.phar/php-code-coverage/CodeCoverage.php:173
    PHP  10. PHP_CodeCoverage->processUncoveredFileFromWhitelist() phar:///usr/local/Cellar/phpunit/5.2.9/libexec/phpunit-5.2.9.phar/php-code-coverage/CodeCoverage.php:670
    PHP  11. include_once() phar:///usr/local/Cellar/phpunit/5.2.9/libexec/phpunit-5.2.9.phar/php-code-coverage/CodeCoverage.php:693

Environment: 
Mac 10.11.3 and Homebrew 0.9.5
Composer version 1.0-dev (f2d606ae0c705907d8bfa1c6f884bced1255b827)
PHPUnit 5.2.9 by Sebastian Bergmann and contributors.
PHP 5.6.18 (cli) (built: Feb  6 2016 06:53:53) 
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies
    with Xdebug v2.3.3, Copyright (c) 2002-2015, by Derick Rethans

The annotations:

    // @codeCoverageIgnore
    /**
     * @codeCoverageIgnore
     */

Or any of the variants of @codeCoverageIgnoreStart @codeCoverageIgnoreEnd with either comment alternatives are ignored and the method tallied into the codeCoverage.

As a side note, the method signature is crashing the code coverage with an error of "method not matching" when the method signature is exactly equal (apart from the abstract / not abstract).

Any idea on how to sort this crash?

**Comment by sebastianbergmann** [comment]
This is not a docblock:
```php
// @codeCoverageIgnore
```

This is a bug in your code:
```
Declaration of datasource_children::SQLUpdateBuild() must be compatible with datasource_sql::SQLUpdateBuild($table, $into, array &$data, $whereSql, $joinSQL = NULL) in /Users/Shared/Jenkins/Home/workspace/dbe/dbEngine/test/Generator/SqlIncTestSqlInc.php on line 4
```

**Comment by lferro9000** [comment]
That is not indeed a docblock, as i stated, i tryed also with the docblock and it didn't worked either.

I've found that systax on the documentation:

https://phpunit.de/manual/current/en/phpunit-book.html#code-coverage-analysis.ignoring-code-blocks

I managed to make it work, i've still no idea why the system / php thought that the method signature was different (when it was a copy of the abstract one without the abstract keywork).

So this is not a blocking issue (for me), and just perhaps a documentation update that is needed.
